src/Nethermind/Nethermind.State/PersistentStorageProvider.cs

```diff
@@ -260,8 +257,34 @@ void UpdateRootHashesSingleThread()
                 }
 
                 StorageTree storageTree = kvp.Value;
-                storageTree.UpdateRootHash(canBeParallel: true);
-                _stateProvider.UpdateStorageRoot(address: kvp.Key, storageTree.RootHash);
+                ref var dict = ref CollectionsMarshal.GetValueRefOrNullRef(_blockChanges, kvp.Key);
+                int writes = 0;
+                int skipped = 0;
+                foreach (var key in dict.Keys)
+                {
+                    ref var change = ref dict.GetValueRefOrNullRef(key);
```

Comment:
Why iterate over keys only then to lookup the value `change`?

Reply:
Chaned